src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +69,11 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> (p.getAge() >= fromAge)
+                        && (p.getSex() == Person.Sex.MAN
+                        ? p.getAge() <= maleToAge : p.getAge() <= femaleToAge))
```

Comment:
```suggestion
         Predicate<Person> ageAndSexFilter = p -> p.getAge() >= fromAge &&
                                                ((p.getSex() == Person.Sex.MAN && p.getAge() <= maleToAge) ||
                                                 (p.getSex() == Person.Sex.WOMAN && p.getAge() <= femaleToAge));
        return peopleList.stream()
                .filter(ageAndSexFilter)
```
but make predicate a class level field

Reply:
I don't understand why this code is more readable... As for me my code is more readable. Why do I have to create Predicate for single using and put it in class field?